src/main/java/appeng/core/features/registries/cell/CellRegistry.java

```diff
@@ -107,4 +110,33 @@ public ICellHandler getHandler( final ItemStack is )
 		}
 		return null;
 	}
+
+	@Override
+	public void addCellGuiHandler( ICellGuiHandler handler )
+	{
+		this.guiHandlers.add( handler );
+	}
+
+	@Override
+	public <T extends IAEStack<T>> ICellGuiHandler getGuiHandler( final IStorageChannel<T> channel, final ItemStack is )
+	{
+		ICellGuiHandler fallBack = null;
+
+		for( final ICellGuiHandler ch : this.guiHandlers )
+		{
+			if( ch.isHandlerFor( channel ) )
+			{
+				if( ch.isSpecializedFor( is ) )
+				{
+					return ch;
+				}
+
+				if( fallBack != null )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2583)